### PR TITLE
Polyhedral

### PR DESCRIPTION
**Pull request by MathieuDutSik**
First version of the compilation of the `polyhedral` code in C++

### CODE REVIEW

File: P/polyhedral/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+name = "POLYHEDRAL"
+version = v"0.1" # <-- This is the first version of it but this is rather arbitrary
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd polyhedral_common
+git submodule update --init --recursive
+cd src_export_oscar
+export GMP_INCDIR=$WORKSPACE/destdir/include
+export GMP_C_LINK="-L$WORKSPACE/destdir/lib -lgmp"
```
Comment (fingolfin) [inline P/polyhedral/build_tarballs.jl:22]:
```suggestion
export GMP_INCDIR=$includedir
export GMP_C_LINK="-L$libdir -lgmp"
```
Also elsewhere

File: P/polyhedral/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export BOOST_LINK="-L$WORKSPACE/destdir/lib -lboost_serialization"
+
+export EIGEN_PATH=$WORKSPACE/destdir/include/eigen3
+
+export NAUTY_INCLUDE="-I$WORKSPACE/destdir/include/nauty"
+export NAUTY_LINK="-L$WORKSPACE/destdir/lib -lnauty"
+
+make
+
+cp GRP_ListMat_Subset_EXT_Automorphism $WORKSPACE/destdir/bin
```
Comment (fingolfin) [inline P/polyhedral/build_tarballs.jl:34]:
```suggestion
cp GRP_ListMat_Subset_EXT_Automorphism $bindir
```
Likewise elsewhere

File: P/polyhedral/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+export GMP_INCDIR=$includedir
+export GMP_C_LINK="-L$libdir -lgmp"
+
+export BOOST_INCDIR=$WORKSPACE/destdir/include
+export BOOST_LINK="-L$WORKSPACE/destdir/lib -lboost_serialization"
+
+export EIGEN_PATH=$WORKSPACE/destdir/include/eigen3
+
+export NAUTY_INCLUDE="-I$WORKSPACE/destdir/include/nauty"
+export NAUTY_LINK="-L$WORKSPACE/destdir/lib -lnauty"
```
Comment (fingolfin) [inline P/polyhedral/build_tarballs.jl:30]:
```suggestion
export BOOST_INCDIR=$includedir
export BOOST_LINK="-L$libdir -lboost_serialization"

export EIGEN_PATH=$includedir/eigen3

export NAUTY_INCLUDE="-I$includedir/nauty"
export NAUTY_LINK="-L$libdir -lnauty"
```

**Comment by fingolfin** [comment]
The failure over FreeBSD seems to be due to a clash between a boost header and a FreeBSD system header:
```
[13:07:36] In file included from ../src_comb/Boost_bitset.h:7,
[13:07:36]                  from ../src_poly/GRP_GroupFct.h:7,
[13:07:36]                  from GRP_ListMat_Subset_EXT_Automorphism.cpp:8:
[13:07:36] ../src_comb/Boost_bitset_kernel.h:60:72: error: macro "setbit" passed 3 arguments, but takes just 2
[13:07:36]    60 | inline void setbit(std::vector<uint8_t> &V, size_t const &pos, bool val) {
[13:07:36]       |                                                                        ^
[13:07:36] In file included from /opt/x86_64-unknown-freebsd12.2/bin/../x86_64-unknown-freebsd12.2/sys-root/usr/local/include/boost/predef/os/bsd/free.h:39,
...
[13:07:36]                  from ../src_number/NumberTheoryBoostGmpInt.h:10,
[13:07:36]                  from GRP_ListMat_Subset_EXT_Automorphism.cpp:4:
[13:07:36] /opt/x86_64-unknown-freebsd12.2/x86_64-unknown-freebsd12.2/sys-root/include/sys/param.h:286: note: macro "setbit" defined here
[13:07:36]   286 | #define setbit(a,i)     (((unsigned char *)(a))[(i)/NBBY] |= 1<<((i)%NBBY))
[13:07:36]       | 
[13:07:36] ../src_comb/Boost_bitset_kernel.h:106:27: error: macro "setbit" passed 3 arguments, but takes just 2
[13:07:36]   106 |         setbit(V, pos, val);
[13:07:36]       |                           ^
```

File: P/polyhedral/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    ExecutableProduct("POLY_cdd_lp2", :POLY_cdd_lp2)
+    ExecutableProduct("POLY_dual_description_group", :POLY_dual_description_group)
+    ExecutableProduct("POLY_sampling_facets", :POLY_sampling_facets)
+    ExecutableProduct("sv_exact", :sv_exact)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("GMP_jll", v"6.2.1"),
+    Dependency("Eigen_jll"),
```
Comment (giordano) [inline P/polyhedral/build_tarballs.jl:65]: I think Eigen is a [`BuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.BuildDependency)?  See also https://docs.binarybuilder.org/stable/build_tips/#Dependencies-for-the-target-system-vs-host-system

File: P/polyhedral/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    Dependency("nauty_jll"; compat = "~2.6.13"),
+    Dependency("boost_jll", compat = "=1.76.0"),
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"12")
```
Comment (giordano) [inline P/polyhedral/build_tarballs.jl:71]: No, GCC 12 is not going to work.  For compatibility you have to use the ***oldest*** version of GCC which can compile your code, not the latest.

Comment (MathieuDutSik) [reply P/polyhedral/build_tarballs.jl]: Ok, and that is VERSION 9.

File: P/polyhedral/build_tarballs.jl (hunk 4 above)
Comment (MathieuDutSik) [reply P/polyhedral/build_tarballs.jl]: OK.

**Pull request merged**